File: 05.1-api-scala/src/main/scala/x/Clickstream.scala
package x

/**
this will process JSON clickstream data

Usage:
spark-submit --class 'x.Clickstream'  target/scala-2.11/testapp_2.11-1.0.jar    <files to process>

*/


import org.apache.spark.SparkContext._
import org.apache.spark.sql.SparkSession

object Clickstream {
  def main(args: Array[String]) {
    if (args.length < 1) {
      println ("need file(s) to load")
      System.exit(1)
    }

    val input = args(0) // can be wildcard  'dir/*.log'

    val spark = SparkSession.builder().
                appName("Clicks -- MYNAME").
                getOrCreate()
    import spark.implicits._

    println ("### Ahoy mate, fix TODO items!")

    // ## TODO-1 : create an input df;
    // hint : spark.read.json(input)
    /*
    val clickstream = ???

    // TODO-2  : count how many records we have
    val count =  ???
    println("### total clickstream records " + count)
    */

    // TODO-3 : count traffic per domain
    /*
    val domainCount = clickstream.groupBy("???").count()

    // display domainCount (hint : show)
    domainCount.???
    */

  }
}

File: java/submit-java/src/main/java/spark/basic/Clickstream.java
package spark.basic;

import org.apache.spark.sql.SparkSession;

/**
Class that process JSON clickstream data
<br>
Usage:
spark-submit --class 'spark.basic.Clickstream' --master spark://localhost:7077 target/spark.basic-0.0.1-SNAPSHOT-jar-with-dependencies.jar  <files to process>

*/
public class Clickstream {

	public static void main(String[] args) {
		if(args.length == 1) {
			String file = args[0];// can be wildcard  'dir/*.log'
			
			// Give a name
			SparkSession spark = SparkSession.builder().appName("Process Files --MyName").master("local").getOrCreate();
			System.out.println ("### Ahoy mate, fix TODO items!");

			// ## TODO-1 : create an input dataframe;
			// hint : spark.read().json(input)
			/*Dataset<Row> clickStream = ???;*/

			// TODO-2  : count how many records we have
			/*Long numberOfRecords = clickStream.???;
			System.out.println("### Total clickstream records " + numberOfRecords);*/

			// TODO-3 : count traffic per domain
			/*Dataset<Row> domainCount = clickStream.groupBy("????").count();
			System.out.println("### Domain count: ");*/
			// display domainCount (hint : show)
			/*domainCount.???;*/
		} else {
			System.out.println ("Need a single argument(file path) to load\nIt can also be a wildcard  \"dir/*.json\"");
			System.exit(1);
		}
	}

}
